### PR TITLE
Allow to define authorized MIME types in ImageManager::validateUpload()

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | * Permit to override the mime-type list in ImageManager::validateUpload<br> * Check the mimetype (if the file is not managed by the previous method in ImageManager::isRealImage
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partial fix for #10600
| How to test?  | Upload doesn't change (sample : Image Product use ImageManager::validateUpload)

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16316)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: classes/ImageManager.php [hunk 1]
```diff
@@ -396,6 +390,26 @@ public static function isRealImage($filename, $fileMimeType = null, $mimeTypeLis
...
+     *
+     * @return bool
+     */
+    public static function isRealImage($filename, $fileMimeType = null, $mimeTypeList = null)
+    {
+        if (!$mimeTypeList) {
+            $mimeTypeList = ['image/gif', 'image/jpg', 'image/jpeg', 'image/pjpeg', 'image/png', 'image/x-png'];
+        }
+
+        $mimeType = self::getMimeType($filename);
```
Comment (PierreRambaud) [inline classes/ImageManager.php]:
```suggestion
        $mimeType = static::getMimeType($filename);
```

File: classes/ImageManager.php [hunk 2]
```diff
@@ -396,6 +390,26 @@ public static function isRealImage($filename, $fileMimeType = null, $mimeTypeLis
...
+     * @param string $filename File path to check
+     * @param string $fileMimeType File known mime type (generally from $_FILES)
+     * @param array<string>|null $mimeTypeList Allowed MIME types
+     *
+     * @return bool
+     */
+    public static function isRealImage($filename, $fileMimeType = null, $mimeTypeList = null)
+    {
+        if (!$mimeTypeList) {
+            $mimeTypeList = ['image/gif', 'image/jpg', 'image/jpeg', 'image/pjpeg', 'image/png', 'image/x-png'];
```
Comment (PierreRambaud) [inline classes/ImageManager.php]: use const instead

File: classes/ImageManager.php (hunk 1 above)
Comment (Progi1984) [reply classes/ImageManager.php]: Thanks

File: classes/ImageManager.php [hunk 3]
```diff
@@ -354,39 +354,33 @@ public static function imagecopyresampled(
...
             if ($imageInfo) {
                 $mimeType = $imageInfo['mime'];
-            } else {
-                $fileMimeType = false;
             }
-        } elseif (function_exists('finfo_open')) {
+        }
+        // Try with FileInfo
+        if (!$mimeType && function_exists('finfo_open')) {
             $const = defined('FILEINFO_MIME_TYPE') ? FILEINFO_MIME_TYPE : FILEINFO_MIME;
```
Comment (PierreRambaud) [inline classes/ImageManager.php]: You can use directly the const, it was for compatiblity with <5.3
```suggestion
            $const = FILEINFO_MIME_TYPE;
```

Comment (Progi1984) [reply classes/ImageManager.php]: :+1:

File: classes/ImageManager.php (hunk 2 above)
Comment (Progi1984) [reply classes/ImageManager.php]: Fixed

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Please, don't forget to add the test part for our QA team :)

**Comment by Progi1984** [comment]
@PierreRambaud :heavy_check_mark: Done

**Comment by PierreRambaud** [comment]
Thanks @Progi1984

**Pull request merged**